specification/web/resource-manager/Microsoft.DomainRegistration/stable/2024-11-01/Domains.json

```diff
@@ -0,0 +1,1616 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2024-11-01",
+    "title": "Domains API Client"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DomainRegistration/checkDomainAvailability": {
+      "post": {
+        "tags": [
+          "Domains"
+        ],
+        "summary": "Check if a domain is available for registration.",
+        "description": "Description for Check if a domain is available for registration.",
+        "operationId": "Domains_CheckAvailability",
+        "parameters": [
+          {
+            "name": "identifier",
+            "in": "body",
+            "description": "Name of the domain.",
+            "required": true,
+            "schema": {
+              "$ref": "../../../Microsoft.Web/stable/2024-11-01/CommonDefinitions.json#/definitions/NameIdentifier"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DomainAvailabilityCheckResult"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "../../../Microsoft.Web/stable/2024-11-01/CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Check domain availability": {
+            "$ref": "./examples/CheckDomainAvailability.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DomainRegistration/domains": {
+      "get": {
+        "tags": [
+          "Domains"
+        ],
+        "summary": "Get all domains in a subscription.",
+        "description": "Description for Get all domains in a subscription.",
+        "operationId": "Domains_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DomainCollection"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "../../../Microsoft.Web/stable/2024-11-01/CommonDefinitions.json#/definitions/DefaultErrorResponse"
```

Comment:
>"$ref": "../../../Microsoft.Web/stable/2024-11-01/CommonDefinitions.json#/definitions/DefaultErrorResponse" [](http://example.com/codeflow?start=14&length=107)

Can you use ErrorResponse from common types? Comment applies throughout this PR.